### PR TITLE
Added support for xterm 256 colors and 24 bit colors

### PR DESCRIPTION
**Pull request by Joe-Merten**
This PR adds support for `esc[38;…m` / `esc[48;…m` for 256 colors and 24 bit foreground / background colors. See also:
- #89
- #44
- https://issues.jenkins-ci.org/browse/JENKINS-24378

Please note, that this needs [jansi PR #72](https://github.com/fusesource/jansi/pull/72) to build & work. I'd increased the jansi version dependency in the `pom.xml` to `1.14-SNAPSHOT` to build locally.

Further, we run into a version bump as described in #95 which leads also to make 256 colors and 24 bit colors don't work (even if we have successfully build and all plugin tests passed).

As I found a way to got it work with Jenkins 1.653 (by just deleting jansi-1.9.jar), I'd found no way to make it work with Jenkins 2.x. As dblock commented in #95, we might have to wait until even the Jenkins guys will pick up the latest jansi changes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,7 +88,7 @@
     <dependency>
       <groupId>org.fusesource.jansi</groupId>
       <artifactId>jansi</artifactId>
-      <version>1.12</version>
+      <version>1.14-SNAPSHOT</version>
```
Comment (dblock) [inline pom.xml]: I don't think we want to ship with an unshipped version of 1.14. I am not saying we shouldn't merge this, but we should be careful releasing with this.

**Comment by dblock** [comment]
... and this doesn't pass build since the SNAPSHOT version is a private one.

**Comment by Joe-Merten** [comment]
Ok, then I just have to wait for:
- [jansi #72](https://github.com/fusesource/jansi/pull/72) became merged
- jansi 1.15 release

Both might take a indefinite period of time.

And after this, I'll:
- change the pom.xml in this PR.
- send a PR to [jenkinsci](https://github.com/jenkinsci/jenkins) to also change their jansi version in pom.xml

CU next time, Joe.

**Comment by dblock** [comment]
Thanks Joe! 

Maybe release what's on master though? It fixes a number of issues.

**Comment by Joe-Merten** [comment]
In #95, you commented:

> … but maybe including JANSI as a built submodule (in a different namespace) is OK?

Could you help me to understand the idea behind this approch?
Then I'll give it a try to make _MoreColors_ work without waiting for Jansi & Jenkins repos (and I could refactor the code as soon as Jansi & Jenkins have been caught up).

**Comment by dblock** [comment]
I mean checking out the source of Jansi as a submodile, then bulk replacing the namespace by something internal during the build and using that code.

**Comment by Joe-Merten** [comment]
I'm a bit aware about git submodule because of trouble in the past (but however, I'm not git expert at all).
What about PR #98? I'd tested it even in bleeding edge jenkins 2.50 and it works as expected.

**Comment by dblock** [comment]
I like it enough. Party on.

**Pull request merged**